structure/heap/heap_test.go

```diff
@@ -0,0 +1,100 @@
+package heap
+
+import (
+	"testing"
+)
+
+type _int int
+
+func (u _int) Less(o _int) bool {
+	return u < o
+}
```

Comment:
Once you use the type constraint you would not need this...

Reply:
I think the type constraint can't be used in many scenarios. The `Less` method can be used in any scenarios.